cdap-watchdog/src/main/java/co/cask/cdap/logging/write/FileMetaDataManager.java

```diff
@@ -147,18 +147,26 @@ public Integer apply(Table table) throws Exception {
             String namespacedLogDir = LoggingContextHelper.getNamespacedBaseDir(logBaseDir, getLogPartition(rowKey));
 
             for (Map.Entry<byte[], byte[]> entry : row.getColumns().entrySet()) {
-              byte[] colName = entry.getKey();
-              if (LOG.isDebugEnabled()) {
-                LOG.debug("Got file {} with start time {}", Bytes.toString(entry.getValue()),
-                          Bytes.toLong(colName));
-              }
-
-              Location fileLocation = locationFactory.create(new URI(Bytes.toString(entry.getValue())));
-              // Delete if file last modified time is less than tillTime
-              if (fileLocation.lastModified() < tillTime) {
-                callback.handle(fileLocation, namespacedLogDir);
-                table.delete(rowKey, colName);
-                deletedColumns++;
+              try {
+                byte[] colName = entry.getKey();
+                URI file = new URI(Bytes.toString(entry.getValue()));
+                if (LOG.isDebugEnabled()) {
+                  LOG.debug("Got file {} with start time {}", file, Bytes.toLong(colName));
+                }
+
+                Location fileLocation = locationFactory.create(file);
+                if (!fileLocation.exists()) {
+                  LOG.debug("File {} does not exist, but metadata is present", file);
```

Comment:
this seems like it should be a warn, since it indicates an inconsistent/unexpected state